src/bootstrap/doc.rs

```diff
@@ -476,11 +476,11 @@ impl Step for Std {
                  .arg("--index-page").arg(&builder.src.join("src/doc/index.md"));
 
             builder.run(&mut cargo);
-            builder.cp_r(&my_out, &out);
         };
         for krate in &["alloc", "core", "std", "proc_macro", "test"] {
             run_cargo_rustdoc_for(krate);
         }
+        builder.cp_r(&my_out, &out);
```

Comment:
Can we avoid this copy entirely by instead symlinking rustdoc's outdir to `out` (the final location)?

Reply:
I haven't strictly tested the symlink, but I wanted to land some easy wins first and start investigating deeper wins later. I believe symlinks won't work because everything attempts to preserve symlinks, and rust-installer wants to actually archive the symlink, which would cause invalid tarballs since we wouldn't actually distribute docs.

I'm working on rust-installer now though so I'll try to plumb this through eventually, ideally I don't want to have to copy documentation anywhere.